### PR TITLE
Add compatibility with 1.34

### PR DESCRIPTION
**Pull request by jyn514**
This PR is best reviewed commit-by-commit. See individual commits for a rationale of the changes.

### CODE REVIEW

**Comment by jyn514** [comment]
I want to add 1.34 to CI but I don't know how to use github actions.

**Comment by jyn514** [comment]
It looks like this doesn't actually run 1.34 and just downloads it temporarily before switching the default to nightly.

**Comment by Kixiron** [comment]
I think the best thing to do is either add a separate msrv job that just runs check on 1.34 or to use [cargo-msrv](https://github.com/foresterre/cargo-msrv) in another job

**Comment by jyn514** [comment]
Testing won't work because criterion isn't compatible with 1.34. The rest is super helpful though!

**Comment by jyn514** [comment]
`--features multi-threaded` also won't work because dashmap isn't compatible with 1.34. This is probably not possible to fix since it uses `alloc` and lasso tries to use `no_std`.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -3,11 +3,38 @@ name: CI
...
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          command: check
+          toolchain: "1.34.0"
+          override: true
+
```
Comment (Kixiron) [inline .github/workflows/main.yml:37]: Is there a reason to have the check job run on 1.34 when there's already a separate MSRV job?

Comment (jyn514) [reply .github/workflows/main.yml:37]: Nope, I forgot to remove it. Good catch!

**Comment by Kixiron** [comment]
Thanks!

**Pull request merged**